databroker/broker.py

```diff
@@ -924,8 +925,10 @@ def wrapped_f(*args, **kwargs):
                 if name == 'descriptor':
                     # Mutate fs_doc here to mark data as external.
                     for data_name in external_writers.keys():
-                        fs_doc['data_keys'][data_name].update(
-                            external='FILESTORE:')
+                        # data doesn't have to exist?
+                        if data_name in fs_doc['data_keys']:
```

Comment:
I don't think this is right. If it's not in `descriptor_doc['data_keys']` then it's not in `event_doc['data']` either, and line 1011 should raise a KeyError.